src/middlewared/middlewared/client/client.py

```diff
@@ -19,10 +18,12 @@
 except ImportError:
     from collections import Callable
 
-from ws4py.client.threadedclient import WebSocketClient
+from websocket import WebSocketApp
+from websocket._abnf import STATUS_NORMAL
+from websocket._http import connect, proxy_info
+from websocket._socket import sock_opt
 
 from . import ejson as json
-from .protocol import DDPProtocol
 from .utils import MIDDLEWARE_RUN_DIR, ProgressBar, undefined
 try:
     from libzfs import Error as ZFSError
```

Comment:
Same with this import, let's fix it since `libzfs` should always be available

Reply:
Someone might theoretically be using this client from a machine where libzfs is not available, so it might have sense to keep it.

Reply:
This code has rarely changed in 5+ years. If you're concerned about 3rd party consumers, then that's something they'll have to patch on their side. We never wrote this to be used outside of our ecosystem. However, if you think we'll break code that is internal then I agree to keep it.